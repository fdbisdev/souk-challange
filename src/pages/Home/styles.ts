import { Platform, StatusBar } from 'react-native';
import styled from 'styled-components/native';
import colors from '../../utils/colors';
import { IButtonProps } from '../../utils/interfaces';

export const Container = styled.SafeAreaView`
  flex: 1;
  padding-top: ${Platform.OS === 'android' ? StatusBar.currentHeight : 0}px;
  background: ${colors.primary};
`;

export const Title = styled.Text`
  font-size: 24px;
  color: ${colors.white};
  font-weight: bold;
  align-self: center;
  margin-top: 5%;
`;

export const ButtonWrapper = styled.View`
  margin-top: 5%;
  flex-direction: row;
  align-items: center;
  justify-content: space-evenly;
`;

export const SelectTypeButton = styled.TouchableOpacity<IButtonProps>`
  border: 1px solid ${colors.white};
  background: ${({ isSelected }) =>
    isSelected ? colors.white : colors.primary};
  padding: 8px;
  width: 40%;
  border-radius: 8px;
`;

export const ButtonText = styled.Text<IButtonProps>`
  font-size: 18px;
  color: ${({ isSelected }) => (isSelected ? colors.black : colors.white)};
  align-self: center;
  font-weight: bold;
`;
