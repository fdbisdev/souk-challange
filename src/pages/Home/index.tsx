import React from 'react';

import { Container, Title } from './styles';

function Home() {
  return (
    <Container>
      <Title>LOGADO</Title>
    </Container>
  );
}

export default Home;
