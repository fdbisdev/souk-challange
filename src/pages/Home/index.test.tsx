import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';
import { fireEvent, act } from '@testing-library/react-native';
import Home from '.';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('Home page', () => {
  it('should useSelector have been called on init', async () => {
    const useSelector = jest.spyOn(reactRedux, 'useSelector');
    useSelector.mockReturnValue({ name: 'test' });
    renderer.create(<Home />);
    expect(useSelector).toHaveBeenCalled();
  });

  it('should render correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should set upcoming movie', () => {
    const component = renderer.create(<Home />);
    const instance = component.root;
    const instanceChild = instance.findByProps({ testID: 'upcomingMovie' });
    expect(instanceChild.props.isSelected).toEqual(true);
    act(() => {
      fireEvent.press(instanceChild);
    });
    expect(instanceChild.props.isSelected).toEqual(false);
  });

  it('should set popular movie', () => {
    const component = renderer.create(<Home />);
    const instance = component.root;
    const instanceChild = instance.findByProps({ testID: 'popularMovie' });
    expect(instanceChild.props.isSelected).toEqual(false);
    act(() => {
      fireEvent.press(instanceChild);
    });
    expect(instanceChild.props.isSelected).toEqual(true);
  });
});
