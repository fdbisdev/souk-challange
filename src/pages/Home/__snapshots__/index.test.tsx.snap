// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#333",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 0,
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#ffffff",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginTop": "5%",
        },
      ]
    }
  >
    Ol√°, 
    test
    !
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginTop": "5%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "width": "40%",
        }
      }
      testID="upcomingMovie"
    >
      <Text
        isSelected={true}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#000000",
              "fontSize": 18,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Upcoming
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#333",
          "borderColor": "#ffffff",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "width": "40%",
        }
      }
      testID="popularMovie"
    >
      <Text
        isSelected={false}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#ffffff",
              "fontSize": 18,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Popular
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "marginTop": "5%",
          "width": "100%",
        },
      ]
    }
  >
    <ActivityIndicator
      color="#ffffff"
      size="large"
    />
  </View>
</RCTSafeAreaView>
`;
